modules/nextflow/src/test/groovy/nextflow/scm/AssetManagerTest.groovy

```diff
@@ -442,6 +445,15 @@ class AssetManagerTest extends Specification {
         dir.resolve('nextflow.config').text = 'manifest {  }'
         dir.resolve('foo.nf').text = 'this is foo content'
 
+        // Use this user's custom defaultBranch name if set in ~/.gitconfig
+        def defaultBranch = 'master'
+        def gitconfig = Paths.get(System.getProperty('user.home'),'.gitconfig');
+        if(gitconfig.exists()) {
+            def config = new Config()
+            config.fromText(gitconfig.text)
+            defaultBranch = config.getString('init', null, 'defaultBranch') ?: 'master'
+        }
+
```

Comment:
Wonder if this is this is really needed. Git should not be aware that nextflow uses `master` as default irrespective of the user config?

Reply:
In the test below, we create a new git repository
```groovy
def init = Git.init()
```
... and the newly created repo will use whatever default branch name the user has set on their machine. The choice of Nextflow to use `master` uses does not affect this test repository, but the git configuration on the machine running the test _does_ affect this repo. 

It's important to know what the user has set as default branch name, because we check the branch name later on in the test. Before this PR, we assume that the ephemeral test repository will have the branch named `master`:
```groovy
script.revision == 'master'
```

Reply:
Fair enough, what about turning that snippet into a helper method `getDefaultBranch()` to keep the test focused on the main logic